opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -99,6 +100,46 @@ def get_context(self) -> 'SpanContext':
         """
 
 
+class TraceOptions(int):
+    """A bitmask that represents options specific to the trace.
+
+    The only supported option is the "recorded" flag (``0x01``). If set, this
+    flag indicates that the trace may have been recorded upstream.
+
+    See the `W3C Trace Context`_ spec for details.
+
+    .. _W3C Trace Context: https://www.w3.org/TR/trace-context/#trace-flags
+    """
+    DEFAULT = 0x00
+    RECORDED = 0x01
+
+    @classmethod
+    def get_default(cls) -> 'TraceOptions':
+        return cls(cls.DEFAULT)
+
+
+DEFAULT_TRACEOPTIONS = TraceOptions.get_default()
+
+
+class TraceState(typing.Dict[str, str]):
+    """A list of key-value pairs that carries system-specific config.
```

Comment:
"system-specific config" is kind of misleading, perhaps we can borrow the wording from W3C spec.

Reply:
This is cribbed from java. I don't see a concise description for this in https://www.w3.org/TR/trace-context/#tracestate-field, what do you recommend?

There's also the issue that this describes the internal struct after reading the header and the W3C spec describes the header itself.

Reply:
From the HTTP spec "The tracestate HTTP header field conveys information about request position in multiple distributed tracing graphs." (we probably need to rephrase and remove HTTP since OpenTelemetry targets more than HTTP protocol).

From the [overview](https://github.com/w3c/trace-context/blob/master/spec/10-overview.md#design-overview) "tracestate extends traceparent with vendor-specific data represented by a set of name/value pairs".

Reply:
"Conveys information about request position in multiple distributed tracing graphs" doesn't explain much.

There's no concept of a traceparent outside the propagator. Is "vendor-specific" instead of "system-specific" clearer?

Reply:
Probably "vendor specific data" instead of "system-specific config"?

I personally would try to avoid this at all, just say something like:
```
"""The implementation of W3C TraceContext standard https://www.w3.org/TR/trace-context/#tracestate-field
"""
```

Reply:
I think we still ought to include _some_ description in the docstring. Updated this in ce95320 to remove most of the text and say "vendor specific info".